### DISCUSSION TITLE
使用infer_det.py预测同一张图片，预测结果不稳定，有随机性，每次都不一样

### DISCUSSION THREAD

**Discussion by gl94** [PaddlePaddle/PaddleOCR#12579] [Q&A]
使用infer_det.py预测同一张图片，打印预测结果的某一行：
![图片](https://github.com/PaddlePaddle/PaddleOCR/assets/26252750/d67a56ea-e060-483d-80a2-00fa648e2547)
进行三次预测，每次预测的结果都不一样：
![图片](https://github.com/PaddlePaddle/PaddleOCR/assets/26252750/92ea37aa-79c3-462f-a9a1-21f8a21fa0d9)
请问这种随机性是如何造成的，如何消除这种随机性？

**Comment by SWHL** [comment]
猜测是忘记开eval模式了，类似torch.eval()操作。
如果问题仍然存在，请拉取最新main分支代码，再次尝试
